### PR TITLE
feat(storage): add projectNumber field to bucketAttrs.

### PR DESCRIPTION
**Pull request by anrajitha**
Fixes: https://github.com/googleapis/google-cloud-go/issues/4804

Adds ProjectNumber field to bucketAttrs.

### CODE REVIEW

File: storage/bucket.go [hunk 1]
```diff
@@ -337,6 +337,9 @@ type BucketAttrs struct {
 	// Typical values are "multi-region", "region" and "dual-region".
 	// This field is read-only.
 	LocationType string
+
+	// The project number of the project the bucket belongs to.
```
Comment (tritone) [inline storage/bucket.go:341]: Note that the field is read-only (as in L339 above).

File: storage/bucket_test.go [hunk 2]
```diff
@@ -642,10 +642,11 @@ func TestNewBucket(t *testing.T) {
...
 		},
-		LocationType: "dual-region",
-		Encryption:   &raw.BucketEncryption{DefaultKmsKeyName: "key"},
-		Logging:      &raw.BucketLogging{LogBucket: "lb", LogObjectPrefix: "p"},
-		Website:      &raw.BucketWebsite{MainPageSuffix: "mps", NotFoundPage: "404"},
+		LocationType:  "dual-region",
+		Encryption:    &raw.BucketEncryption{DefaultKmsKeyName: "key"},
+		Logging:       &raw.BucketLogging{LogBucket: "lb", LogObjectPrefix: "p"},
+		Website:       &raw.BucketWebsite{MainPageSuffix: "mps", NotFoundPage: "404"},
+		ProjectNumber: 123231313,
```
Comment (tritone) [inline storage/bucket_test.go:649]: Let's also add a check to the integration test for buckets to ensure that this is received. https://github.com/googleapis/google-cloud-go/blob/81fda19b595a4fab7d2ff322d8524d2c7aced16e/storage/integration_test.go#L255

**Review by tritone — COMMENTED** [review]
Generally looks good!

File: storage/bucket.go (hunk 1 above)
Comment (anrajitha) [reply storage/bucket.go:341]: Done. Added the comment.

File: storage/bucket_test.go (hunk 2 above)
Comment (anrajitha) [reply storage/bucket_test.go:649]: Done., added a check on projectNumber

File: storage/integration_test.go [hunk 3]
```diff
@@ -265,6 +265,9 @@ func TestIntegration_BucketMethods(t *testing.T) {
 	if attrs.LocationType == "" {
 		t.Error("got an empty LocationType")
 	}
+	if attrs.ProjectNumber == 0 {
+		t.Errorf("got a zero ProjectNumber")
+	}
```
Comment (BrennaEpp) [inline storage/integration_test.go:270]: Can we add this check for creating/deleting with attributes in this same test as well?

Comment (anrajitha) [reply storage/integration_test.go:270]: Done. Added this check in there as well.

**Review by BrennaEpp — APPROVED** [review]
LGTM

**Pull request merged**